plugins/ccgx/fu-ccgx-common.c

```diff
@@ -0,0 +1,96 @@
+	{"CCG3", "CYPD3121-40LQXI",		0x1D0211AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3122-40LQXI",		0x1D0311AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3125-40LQXI",		0x1D0411AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3135-40LQXI",		0x1D0511AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3135-16SXQ'",		0x1D0611AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3126-42FNXI",		0x1D0711AD, 0x80,  0x20000},
+	{"CCG3", "CYPD3123-40LQXI",		0x1D0911AD, 0x80,  0x20000},
+	{"CCG4", "CYPD4225-40LQXI",		0x180011A8, 0x100, 0x20000},
+	{"CCG4", "CYPD4125-40LQXI",		0x180111A8, 0x100, 0x20000},
+	{"CCG4", "CYPD4235-40LQXI",		0x180211A8, 0x100, 0x20000},
+	{"CCG4", "CYPD4135-40LQXI",		0x180311A8, 0x100, 0x20000},
+	{"CCG4", "CYPD4225A0-33FNXIT",		0x181011A8, 0x100, 0x20000},
+	{"CCG4", "CYPD4226-40LQXI",		0x1F0011AF, 0x100, 0x20000},
+	{"CCG4", "CYPD4126-40LQXI",		0x1F0111AF, 0x100, 0x20000},
+	{"CCG4", "CYPD4126-24LQXI",		0x1F0411AF, 0x100, 0x20000},
+	{"CCG4", "CYPD4236-40LQXI",		0x1F0211AF, 0x100, 0x20000},
+	{"CCG4", "CYPD4136-40LQXI",		0x1F0311AF, 0x100, 0x20000},
+	{"CCG4", "CYPD4136-24LQXI",		0x1F0511AF, 0x100, 0x20000},
+	{"CCG3PA", "CYPD3174-24LQXQ",		0x200011B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3174-16SXQ",		0x200111B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3175-24LQXQ",		0x200211B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3171-24LQXQ",		0x200311B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3195-24LDXS",		0x200511B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3196-24LDXS",		0x200611B0, 0x80,  0x10000},
+	{"CCG3PA", "CYPD3197-24LDXS",		0x200711B0, 0x80,  0x10000},
+	{"CCG3PA2", "CYPDC1185-32LQXQ",		0x240011B4, 0x80,  0x20000},
+	{"CCG3PA2", "CYPDC1186-30FNXI",		0x240111B4, 0x80,  0x20000},
+	{"CCG3PA2", "CYPDC1186B2-30FNXI",	0x240211B4, 0x80,  0x20000},
+	{"CCG5", "CYPD5225-96BZXI",		0x210011B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5125-40LQXI",		0x210111B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5235-96BZXI",		0x210211B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5236-96BZXI",		0x210311B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5237-96BZXI",		0x210411B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5227-96BZXI",		0x210511B1, 0x100, 0x20000},
+	{"CCG5", "CYPD5135-40LQXI",		0x210611B1, 0x100, 0x20000},
+	{"CCG6", "CYPD6125-40LQXI",		0x2A0011BA, 0x100, 0x20000},
+	{"CCG6", "CYPD6126-96BZXI",		0x2A1011BA, 0x100, 0x20000},
+	{"CCG6", "CYPD5126-40LQXI",		0x2A0111BA, 0x100, 0x20000},
+	{"CCG6", "CYPD5137-40LQXI",		0x2A0211BA, 0x100, 0x20000},
+	{"CCG6", "CYPD6137-40LQXI",		0x2A0311BA, 0x100, 0x20000},
+	{"PAG1S", "CYPAS111-24LQXQ",		0x2B0111BB, 0x80,  0x10000},
+	{"PAG1S", "CYPD3184-24LQXQ",		0x2B0011BB, 0x80,  0x10000},
+	{"HX3PD", "CYUSB4347-BZXC_PD",		0x1F8211AF, 0x100, 0x20000},
+	{"ACG1F", "CYAC1126-24LQXI",		0x2F0011BF, 0x40,  0x4000},
+	{"ACG1F", "CYAC1126-40LQXI",		0x2F0111BF, 0x40,  0x4000},
+	{"CCG6DF", "CYPD6227-96BZXI",		0x300011C0, 0x80,  0x10000},
+	{"CCG6DF", "CYPD6127-96BZXI",		0x300111C0, 0x80,  0x10000},
+	{"CCG6SF", "CYPD6128-96BZXI",		0x330011C3, 0x80,  0x10000},
+	{"CCG6SF", "CYPD6127-48LQXI",		0x330111C3, 0x80,  0x10000},
```

Comment:
This seems like the kind of data that belongs in a quirk file.

Reply:
The table contains silicon id, flash row size, flash size of each CCGx product. we use this table to find flash row size and flash size via silicon id from cyacd (cypress firmware file) and device to calculate meta data location and information.

Could you let me know how to put this table into quirik file and get flash row size, flash size via silicon id?

Reply:
Sure, I can help with that.

Reply:
Please let me know how to do it.

Reply:
I had a go at this over the weekend. Don't worry about this, I'll include it in my cleanup.